presto-main/src/main/java/io/prestosql/sql/planner/DistributedExecutionPlanner.java

```diff
@@ -114,10 +121,27 @@ private StageExecutionPlan doPlan(SubPlan root, Session session, ImmutableList.B
             dependencies.add(doPlan(childPlan, session, allSplitSources));
         }
 
+        // extract TableInfo
+        Map<PlanNodeId, TableInfo> tables = new HashMap<>();
+        root.getFragment().getRoot().accept(
```

Comment:
Maybe you could use `io.prestosql.sql.planner.optimizations.PlanNodeSearcher#searchFrom(io.prestosql.sql.planner.plan.PlanNode, io.prestosql.sql.planner.iterative.Lookup)`?

```
searchFrom( root.getFragment().getRoot())
  .where(TableScanNode.class::isInstanceOf)
  .findAll()
  .stream()
  .collect(toImmutableMap(...));
```